cloudcix_primitives/cpu_lxd.py

```diff
@@ -1,13 +1,118 @@
+    :param project: The LXD project name.
+    :param name: The name of the LXD instance.
+    :param instance_type: The type of the LXD instance, either "containers" or "virtual_machines".
+    :param cpu: The number of CPU cores to set.
+    :param verify_lxd_certs: Boolean to verify LXD certs.
+    :return: A tuple with a boolean flag indicating success or failure, a message, and a dictionary of successful payloads.
+    """
+    # Define message
+    messages = {
+        1000: f'Successfully updated the CPU limit for {instance_type} {name} on {endpoint_url}',
+        3011: f'Invalid instance_type "{instance_type}" sent. Supported instance types are "containers" and "virtual_machines"',
+        3021: f'Failed to connect to {endpoint_url} for {instance_type}.get payload',
+        3022: f'Failed to run {instance_type}.get payload on {endpoint_url}. Payload exited with status ',
+        3023: f'Failed to update the CPU limit. Error: ',
+        3423: f'Failed to quiesce {instance_type} on {endpoint_url}. Instance was found in an unexpected state of ',
+        3025: f'Failed to set CPU limit for {instance_type} {name}',
+        3523: f'Failed to restart {instance_type} on {endpoint_url}. Instance was found in an unexpected state of ',
+        'default': 'An unexpected error occurred. Error code: ',
+    }
+
+    # validation
+    if instance_type not in SUPPORTED_INSTANCES:
+        return False, f'3011: {messages[3011]}', {}
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        # Get the instance
+        ret = rcc.run(cli=f'{instance_type}.get', name=name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[3021]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[3022]}"), fmt.successful_payloads
+
+        instance = ret['payload_message']
+        fmt.add_successful(f'{instance_type}.get', ret)
+
+        # Quiesce the instance
+        state = instance.state()
+        if state.status == 'Running':
+            instance.stop(force=False, wait=True)
+        elif state.status != 'Stopped':
+            return False, f"{prefix+3}: {messages[3423]} {state.status}", fmt.successful_payloads
+            
+        # Update the CPU limit
+        try:
+            instance.config['limits.cpu'] = str(cpu)
+            instance.save(wait=True)
+            fmt.add_successful(f'{instance_type}.set', {'limits.cpu': str(cpu)})
+        except Exception as e:
+            return False, f"{prefix+4}: {messages[3025]}: {e}", fmt.successful_payloads
+
+        # Restart the instance
+        try:
```

Comment:
This can also be removed as the driver, if needed the driver will call the lxd.restart() primitive. 

It could be the case that the customer
1. Quiesced the LXD instance  (LXD Driver quiesce)
2. Changed CPU/RAM  (LXD Driver update_quiesced)

As the primitive does not know the state of the LXD instance in the IaaS application, it should not stop/restart. 

By starting here, the instance on the host will not be the same as in the IaaS database which is why it is the drivers responsibility to manage quiescing and restarting